### DISCUSSION TITLE
Evaluate using FUSE (FileSystem in Userspace) mounting to reduce disk writes

### DISCUSSION THREAD

**Discussion by trivikr** [yarnpkg/berry#4022] [Ideas]
Usage of FUSE was mentioned in [In-depth of tnpm rapid mode](https://dev.to/atian25/in-depth-of-tnpm-rapid-mode-how-could-we-fast-10s-than-pnpm-3bpp) blog post.

From the blog post
> nydus doesn't support macOS at the moment, so we implemented an adapter of nydus to macfuse. We'll open source it when it's ready.

This blog post was discussed on Discord, starting with the following comment:
> [Fuse was considered a very long time ago, back when I pitched PnP for Yarn 1](https://discord.com/channels/226791405589233664/226791405589233664/935662736334008320)

Example LICENSE update issue with osxfuse https://github.com/osxfuse/osxfuse/issues/616

**Comment by killagu** [comment]
I’m very glad to help yarn to use fuse for performance.

The nydus will support macOS before 2022 q2.
